File: medium/78. Subsets/78. Subsets.ts
function subsets(nums: number[]): number[][] {
    const res: number[][] = []

    const backtrack = (path: number[], arr: number[]) => {
        res.push(path)

        if (!res.length) return
        for (let i = 0; i < arr.length; i++) {
            backtrack(path.concat([arr[i]]), arr.slice(i + 1,))
        }
    }

    backtrack([], nums)

    return res
};

File: medium/78. Subsets/78. Subsets.py
from typing import List


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res: List[List[int]] = []

        def backtrack(path: List[int], arr: List[int]):
            res.append(path)

            if not arr:
                return

            for i in range(len(arr)):
                backtrack(path + [arr[i]], arr[i + 1 :])

        backtrack([], nums)

        return res


solution = Solution()
print(solution.subsets(nums=[1, 2, 3]))
